secret <file> is not parsed appropriately

openvpn_server_config_parser.exe static-home.conf 
openvpn_server_config_parser.exe: [INFO] Reading file "static-home.conf"
openvpn_server_config_parser.exe: [WARNING] IGNORING "up" "./home.up"
openvpn_server_config_parser.exe: [ERROR] : Error at byte offset 607: "secret static.key\n\n# Cipher to use\ncipher AES-256-CBC\n\n# OpenVPN 2.0 uses UDP port 1194 by default\n#"

--- a/src/config.ml
+++ b/src/config.ml
@@ -892,6 +892,12 @@ let a_secret str =
   | Ok (a, b, c, d) -> Ok (B (Secret, (a, b, c, d)))
   | Error e -> Error e
 
+let a_secret_maybe_inline =
+  a_option_with_single_path "secret" () >>| fun source ->
+  match source with
+  | `Need_inline () -> `Need_inline `Secret
+  | `Path (path, ()) -> `Path (path, `Secret)
+
 let a_ign_whitespace_nl = skip_many (a_newline <|> a_whitespace_unit)
 
 let a_auth_user_pass_payload =
@@ -1441,6 +1447,7 @@ let a_config_entry : line A.t =
          a_topology;
          a_tls_ciphersuite;
          a_tls_cipher;
+         a_secret_maybe_inline;
          a_not_implemented;
          a_whitespace *> return (`Ignored "");
        ]

--- a/src/config.ml
+++ b/src/config.ml
@@ -887,12 +887,12 @@ let a_tls_ciphersuite =
 let a_auth_user_pass =
   a_option_with_single_path "auth-user-pass" `Auth_user_pass
 
-let a_secret str =
+let a_secret_payload str =
   match parse_string ~consume:Consume.All (inline_payload "secret") str with
   | Ok (a, b, c, d) -> Ok (B (Secret, (a, b, c, d)))
   | Error e -> Error e
 
-let a_secret_maybe_inline =
+let a_secret =
   a_option_with_single_path "secret" () >>| fun source ->
   match source with
   | `Need_inline () -> `Need_inline `Secret
@@ -1447,7 +1447,7 @@ let a_config_entry : line A.t =
          a_topology;
          a_tls_ciphersuite;
          a_tls_cipher;
-         a_secret_maybe_inline;
+         a_secret;
          a_not_implemented;
          a_whitespace *> return (`Ignored "");
        ]
@@ -1506,7 +1506,7 @@ let parse_inline str = function
   | `Tls_key -> a_key_payload str
   | `Tls_crypt_v2 force_cookie ->
     parse_string ~consume:Consume.All (a_tls_crypt_v2_payload force_cookie) str
-  | `Secret -> a_secret str
+  | `Secret -> a_secret_payload str
   | kind ->
       Error
         ("config-parser: not sure how to parse inline "

--- a/test/config_tests.ml
+++ b/test/config_tests.ml
@@ -684,8 +684,10 @@ let tests =
     ( "parsing configuration 'client'",
       `Quick,
       parse_client_configuration ~config:client_conf "client.conf" );
-    (* "parsing configuration 'static-home'", `Quick,
-       parse_client_configuration "static-home.conf" ; -- secret static.key *)
+    ( "parsing configuration 'static-home'", `Quick,
+       parse_client_configuration "static-home.conf" ) ;
+    ( "parsing configuration 'static-home-inline-secret'", `Quick,
+       parse_client_configuration "static-home-inline-secret.conf" ) ;
     ( "parsing configuration 'tls-home'",
       `Quick,
       parse_client_configuration ~config:tls_home_conf "tls-home.conf" );

--- a/test/sample-configuration-files/static-home-inline-secret.conf
+++ b/test/sample-configuration-files/static-home-inline-secret.conf
@@ -1,0 +1,99 @@
+#
+# Sample OpenVPN configuration file for
+# home using a pre-shared static key.
+#
+# '#' or ';' may be used to delimit comments.
+
+# Use a dynamic tun device.
+# For Linux 2.2 or non-Linux OSes,
+# you may want to use an explicit
+# unit number such as "tun1".
+# OpenVPN also supports virtual
+# ethernet "tap" devices.
+dev tun
+
+# Our OpenVPN peer is the office gateway.
+remote 1.2.3.4
+
+# 10.1.0.2 is our local VPN endpoint (home).
+# 10.1.0.1 is our remote VPN endpoint (office).
+ifconfig 10.1.0.2 10.1.0.1
+
+# Our up script will establish routes
+# once the VPN is alive.
+up ./home.up
+
+# Our pre-shared static key
+secret [inline]
+
+# Cipher to use
+cipher AES-256-CBC
+
+# OpenVPN 2.0 uses UDP port 1194 by default
+# (official port assignment by iana.org 11/04).
+# OpenVPN 1.x uses UDP port 5000 by default.
+# Each OpenVPN tunnel must use
+# a different port number.
+# lport or rport can be used
+# to denote different ports
+# for local and remote.
+; port 1194
+
+# Downgrade UID and GID to
+# "nobody" after initialization
+# for extra security.
+; user nobody
+; group nobody
+
+# If you built OpenVPN with
+# LZO compression, uncomment
+# out the following line.
+; comp-lzo
+
+# Send a UDP ping to remote once
+# every 15 seconds to keep
+# stateful firewall connection
+# alive.  Uncomment this
+# out if you are using a stateful
+# firewall.
+; ping 15
+
+# Uncomment this section for a more reliable detection when a system
+# loses its connection.  For example, dial-ups or laptops that
+# travel to other locations.
+; ping 15
+; ping-restart 45
+; ping-timer-rem
+; persist-tun
+; persist-key
+
+# Verbosity level.
+# 0 -- quiet except for fatal errors.
+# 1 -- mostly quiet, but display non-fatal network errors.
+# 3 -- medium output, good for normal operation.
+# 9 -- verbose, good for troubleshooting
+verb 3
+
+<secret>
+#
+# 2048 bit OpenVPN static key
+#
+-----BEGIN OpenVPN Static key V1-----
+87055d27a5536ac72e129916f4287adb
+fce68b7ef6d929539ea170ed0ddf6822
+899f5dbe6aa5df17673c10d63bfe5221
+a25824527c60187666406d92c18dfc3a
+ec597ed09c5aaacc2256c2303e71e17e
+ff995ce7760877abee1d400ea768ace6
+3dcc7d0ef10f1f6d4df4822a78ebbf87
+99e1ddcf2e206872235eb7a92fddd560
+99654cb6d0d19dc099fdfe318382c5b8
+f508feaf3818d8bb35d0afea0e609681
+8d7eaf4dc8ee072188c414405d6a0ec7
+079d4faaf8520e77eee535e4cc0c7785
+5f70cc929d9b5fcbab6e939c088962e4
+7fe05b2e4367c15ddf8f1824b7d772a6
+668345bc7b2f847d03080abb59ff37f2
+1f7c6528d77584af997c0779a1c7e36f
+-----END OpenVPN Static key V1-----
+</secret>

